sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -452,91 +453,103 @@ pi_result OCL(piEnqueueMemBufferMap)(
   return ret_err;
 }
 
-// TODO: Remove the 'OclPtr' extension used with the PI_APIs.
-// Forward calls to OpenCL RT.
+pi_result piPluginInit(pi_plugin *PluginInit) {
+  strcpy(PluginInit->PluginVersion, SupportedVersion);
```

Comment:
As the version array is of fixed size the use of [`strncpy`](https://en.cppreference.com/w/cpp/string/byte/strncpy) would be safer here.

Wondering though if it would be ok to make `PluginInit->PluginVersion` a `const char*` to point to the version string literal in this file?

Reply:
Hmmm..
I thought keeping all memory allocation and destruction in Plugin Interface will be a better idea. The library is loaded and it populates the fields.
I will take the focus off the version discussion and add a TODO for future.

Reply:
What I mean it, if the plugin cpp file defines the version as a string literal, then `piPluginInit` could just point `PluginInit->PluginVersion` to the string literal. No copying required and therefore no memory management.

There might be intricacies with the plugin unloading though which could leave a dangling pointer but my guess is that an unloaded plugin means that most SYCL calls become unsafe anyway.